xclim/indices/generic.py

```diff
@@ -594,3 +594,56 @@ def extreme_temperature_range(
     u = str2pint(low_data.units)
     out.attrs["units"] = pint2cfunits(u - u)
     return out
+
+
+def aggregate_between_dates(data, start, end, op="sum", freq=None):
```

Comment:
I think we should add annotations here (`data: xr.DataArray` and the such), I know we don't do it everywhere, but the goal would be to have them on all functions with a potential to be used externally.

Reply:
I agree.  This could end up being a pretty useful function for expert users who want to customize the resampling period(s)